src/Sentry.Unity/Integrations/UnityApplicationLoggingIntegration.cs

```diff
@@ -0,0 +1,98 @@
+        internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
+
+        // TODO: remove 'IEventCapture' in  further iteration
+        private readonly IEventCapture? _eventCapture;
+        private readonly IAppDomain _appDomain;
+
+        private IHub? _hub;
+
+        public UnityApplicationLoggingIntegration(IAppDomain? appDomain = null, IEventCapture? eventCapture = null)
+        {
+            _appDomain = appDomain ?? UnityAppDomain.Instance;
+            _eventCapture = eventCapture;
+        }
+
+        public void Register(IHub hub, SentryOptions _)
+        {
+            _hub = hub;
+
+            _appDomain.LogMessageReceived += OnLogMessageReceived;
+            _appDomain.Quitting += OnQuitting;
+        }
+
+        // Internal for testability
+        internal void OnLogMessageReceived(string condition, string stackTrace, LogType type)
+        {
+            var debounced = type switch
+            {
+                LogType.Error or LogType.Exception or LogType.Assert => ErrorTimeDebounce.Debounced(),
+                LogType.Log => LogTimeDebounce.Debounced(),
+                LogType.Warning => WarningTimeDebounce.Debounced(),
+                _ => true
+            };
+            if (!debounced)
+            {
+                return;
+            }
+
+            // TODO: to check against 'MinBreadcrumbLevel'
+            if (type != LogType.Error && type != LogType.Exception && type != LogType.Assert)
+            {
+                // TODO: MinBreadcrumbLevel
+                // options.MinBreadcrumbLevel
+                _hub?.AddBreadcrumb(condition, level: ToBreadcrumbLevel(type));
+                return;
+            }
+
+            var sentryEvent = new SentryEvent(new UnityLogException(condition, stackTrace));
+            sentryEvent.SetTag("log.type", ToEventTagType(type));
+
+            _eventCapture?.Capture(sentryEvent); // TODO: remove, for current integration tests compatibility
+            _hub?.CaptureEvent(sentryEvent);
+
+            _hub?.AddBreadcrumb(condition, level: ToBreadcrumbLevel(type));
+        }
+
+        private void OnQuitting()
+        {
+            // Note: iOS applications are usually suspended and do not quit. You should tick "Exit on Suspend" in Player settings for iOS builds to cause the game to quit and not suspend, otherwise you may not see this call.
+            //   If "Exit on Suspend" is not ticked then you will see calls to OnApplicationPause instead.
+            // Note: On Windows Store Apps and Windows Phone 8.1 there is no application quit event. Consider using OnApplicationFocus event when focusStatus equals false.
+            // Note: On WebGL it is not possible to implement OnApplicationQuit due to nature of the browser tabs closing.
+            Application.logMessageReceived -= OnLogMessageReceived;
+            SentrySdk.Close();

```

Comment:
```suggestion
            _hub?.Dispose();
```

One downside of us closing the SDK here (which happened in aspnetcore) is that the order of stuff unregistering is unclear so we can turn off the SDK and more code running ends up on an error that we don't capture.

Alternatively we could simply Flush here (though we only have async flush so would require `GetAwaiter().GetResult()` here)

Reply:
https://github.com/getsentry/sentry-unity/issues/141

Reply:
I can't call `Dispose` on `IHub`.

Reply:
```suggestion
            (_hub as IDisposable)?.Dispose()
```

Reply:
So, `Sentry.Internal.Hub` inherits from `IDisposable` but not `IHub` itself. Is this some kind of internal contract? Why `IHub` itself is not `IDisposable`?

For example, if I decide to create `AnotherHub` (not test one) which inherits from `IHub`, do I need to also inherit from `IDisposable`? Is this implicit contract each hub should abide to?